### DISCUSSION TITLE
Equivalent of `npm run <script>`?

### DISCUSSION THREAD

**Discussion by lowlighter** [denoland/deno#9753] [Q&A] [answered]
Could not find any info on it, but I'd like to run aliased commands like `deno run-script start` or `deno run-script publish` without needed to specify `deno run` with correct permissions, flags, import map and entry point each time.

Is there some kind of an equivalent of `npm run <script>` in deno cli ?

If not, is there any plans to have something like this in the future, or has it been ruled out/already discussed and we need to rely on third party modules?

**Comment by buttercubz** [comment, marked as answer]
> Could not find any info on it, but I'd like to run aliased commands like `deno run-script start` or `deno run-script publish` without needed to specify `deno run` with correct permissions, flags, import map and entry point each time.
> 
> Is there some kind of an equivalent of `npm run <script>` in deno cli ?
> 
> If not, is there any plans to have something like this in the future, or has it been ruled out/already discussed and we need to rely on third party modules?

there are some cli that solve these problems:

https://deno.land/x/trex@v1.5.0#run-scripts-experimental
https://deno.land/x/denox@0.4.0
https://deno.land/x/velociraptor@1.0.0-beta.17

**Reply by lowlighter** [reply]
Velociraptor is very nice!

**Comment by lucacasonato** [comment]
You can use a [`makefile`](https://opensource.com/article/18/8/what-how-makefile).

**Comment by garretmh** [comment]
Deno 1.20 added (unstable) task runner support: https://deno.land/manual/tools/task_runner

The quickest way to try it out is to create a `deno.json` file in your root with a "tasks" entry:
```json
{
  "tasks": {
    "start": "deno run ./main.ts"
  }
}
```
And run those tasks with with:
```sh
deno task start
```
